agent/src/agent/pipeline/manager.py

```diff
@@ -331,6 +335,13 @@ def increase_retry_counter(pipeline_: Pipeline):
     pipeline.repository.save(pipeline_.retries)
 
 
+def set_pipeline_retries_notification_sent(pipeline_: Pipeline, value: bool):
+    if not pipeline_.retries:
```

Comment:
I think it should not be possible to call this method when retries don't exist, so this check is probably not needed
Or at least you don't need to create retries if they don't exist